[Lethicia Renissa] iP

lethiciars opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's, 
* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And its __FREE__!

Features:
- [x] Managing Tasks
- [x] Managing deadlines (coming soon) 
- [ ] Reminders (coming soon)
***
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

lethiciars committed:
Add Greet,Echo,Exit

lethiciars committed:
Add add,list

lethiciars committed:
Add mark done

lethiciars committed:
Add todo,event,deadline

lethiciars committed:
Add Testing

lethiciars committed:
Add error handling

lethiciars committed:
Add Delete

lethiciars committed:
Add methods in Duke

lethiciars committed:
Add documentations

lethiciars committed:
Update test cases

lethiciars committed:
Add saving and writing to file, enums

lethiciars committed:
Fix bugs for todo command

lethiciars committed:
Add javadocs

lethiciars committed:
Add javadocs

lethiciars committed:
Add dates for deadline

lethiciars committed:
Add date feature to deadline

lethiciars committed:
Refactoring Duke to include more oop

lethiciars committed:
Add Command class

lethiciars committed:
Divide classes into packages

lethiciars committed:
Delete unecessary method

lethiciars committed:
Add JUnit Tests

lethiciars committed:
Add javadoc

lethiciars committed:
Add javadoc

lethiciars committed:
Edit to match coding standard

lethiciars committed:
Edit according to coding standard

lethiciars committed:
Add find feature

lethiciars committed:
Add find feature

lethiciars committed:
Fix parser bugs

lethiciars committed:
Update find method

lethiciars committed:
Merge branches 'master' and 'add-gradle-support' of https://github.com/lethiciars/ip

squeakysquak commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+     */
+    public void run() {
+        Ui.showWelcomeMessage();
+        Scanner scan = new Scanner(System.in);
```
This is very good coding standard adherance. The spaces between the equals sign is very nice.

squeakysquak commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,18 @@
+package duke;
+
+/**
+ * An Exception class for Duke specific Exceptions.
```
This is very good coding standard adherance. The javadoc comment is very nice.

squeakysquak commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,102 @@
+     */
+    public static Command getCommand(String input) throws DukeException {
+        if (input.trim().equals("bye")){
+            return new ExitCommand();
```
This is very good coding standard adherance. The egyptian brackets and indentation is very nice.

squeakysquak reviewed (commented):
This is very good coding standard adherance.

lethiciars committed:
Add GUI

lethiciars committed:
Add GUI

lethiciars committed:
Edit return statements

lethiciars committed:
Update GUI

lethiciars committed:
Change GUI font

lethiciars committed:
Merge branch-Level-10

lethiciars committed:
Add checkstyle

bklimey commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+    public String toDisplay() {
+        String str = "";
+        int i = 1;
+        for (Task item:TaskList) {
```
Minor nit: Maybe [colon could be surrounded by white space](https://se-education.org/guides/conventions/java/intermediate.html#layout)?
`Layout` > `White space within a statement`

luluyousef commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @throws DukeException
+     */
+    @Override
+    public String execute(TaskList taskList){
```
I think you might've missed the space before the braces, can add them to follow coding standard :)

bklimey commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,40 @@
+ * A DeleteCommand class encapsulates the instructions for Duke to delete a task.
+ */
+public class DeleteCommand extends Command {
+    private int TaskNo;
```
Maybe `TaskNo` could be in camelCase (i.e. `taskNo`)?

From [guildeline](https://se-education.org/guides/conventions/java/intermediate.html#naming):
> Variable names must be in camelCase.

luluyousef commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,43 @@
+
+import java.util.ArrayList;
+
+public class FindCommand extends Command{
```
I think you might've missed the space after Command too, can add them to follow coding standard :)

bklimey replied:
Minor suggestion: Maybe `CommandFinder` instead of `FindCommand` since 'Find' in this case is a verb?

From [guideline on naming](https://se-education.org/guides/conventions/java/intermediate.html#naming):
> Class/enum names must be nouns and written in PascalCase.

luluyousef commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,37 @@
+     * @throws DukeException
+     */
+    @Override
+    public String execute(TaskList taskList){
```
can also add space before braces here to follow coding standard :)

luluyousef commented:
Overall, your code looks very readable to me and follows the coding conventions well! Maybe try to fix the spacing before braces in your code to follow the coding conventions more accurately :) Good job!!!

bklimey commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    /**
+     * returns the type of command
```
Conventionally, the description starts with a cap and ends with a fullstop (i.e. `Returns the type of command.`).

bklimey commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,130 @@
+     */
+    public void addTask(taskType type, String[] details) {
+        switch (type) {
+        case TODO:
```
I like that you used a switch statement to keep the code neat! 👍

bklimey commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,102 @@
+                String str = "Please use the format done <task No.>";
+                throw new DukeException(str);
+            }
+        } else if (input.matches("delete (.*)")) {
```
I like your use of regular expression! 👍

bklimey commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,102 @@
+                String validInput = input.substring(start);
+                String taskDesc = validInput.substring(6);
+                String[] fields = taskDesc.split(" /at ", 2);
+                //make sure not empty
```
Maybe rephrase as `Ensure that it is not empty`? Additionally, maybe it could be more descriptive (e.g. what is not empty)?

From [guildeline on comments](https://se-education.org/guides/conventions/java/intermediate.html#comments):
>All comments should be written in English.
>Furthermore, use American spelling and avoid local slang.
 >Rationale: The code is meant for an international audience.

bklimey reviewed (commented):
It is a pleasure to read the code as they are neat and tidy, and there are good use of descriptive names. Generally, I did not find any major issue. Keep it up! :)

lethiciars committed:
Add assertions

lethiciars committed:
Add Assertions to classes

Added assertions where necessary so that the runtime can verify certain
assumptions about the program state.

Classes where assertions were added:
1. Duke
2. Storage
3. Parser
4. TaskList

lethiciars committed:
Merge pull request #1 from lethiciars/branch-A-Assertions

Add assertions to classes

lethiciars committed:
Resolved checkstyle

lethiciars committed:
Fix code to improve code quality

Refactor some classes to improve readability and to remove coding
standards violations.

Some of the changes include:
1. Refactor Parser getCommand method to separate methods
2. Add default branch in case statements

lethiciars committed:
Merge pull request #2 from lethiciars/branch-A-CodeQuality

Fix code to improve code quality

Refactor some classes to improve readability and to remove coding
standards violations.

Some of the changes include:
1. Refactor Parser getCommand method to separate methods
2. Add default branch in case statements

lethiciars committed:
Add C-Help extension

Provide in-App guidance to users upon typing 'help'.

The help command displays a list of commands that can be used on Duke
and te required format.

lethiciars committed:
Merge pull request #3 from lethiciars/branch-C-Help

Add C-Help extension 

Provide in-App guidance to users upon typing 'help'.

lethiciars committed:
Fix javadoc

lethiciars committed:
Improve GUI

Improve the GUI to make it more polished and pleasing to look at.

Changes made:
- Tweak the GUI to match the asymmetric nature of the conversation
ie remove chat bubble for Duke's responses to differentiate the visual
format between user and application messages.
- Tweak the GUI to optimize for space ie make profile picture smaller.

lethiciars committed:
Add product screenshot

lethiciars committed:
Update User Guide

lethiciars committed:
Set theme jekyll-theme-cayman

lethiciars committed:
Set theme jekyll-theme-cayman

lethiciars committed:
Add equals method for testing

lethiciars committed:
Merge branch 'master' of https://github.com/lethiciars/ip
Merge with remote

lethiciars committed:
Add JUnit tests

lethiciars committed:
Add more Junit tests

lethiciars committed:
Update duke.txt

lethiciars committed:
Revert commit

lethiciars committed:
Add message for empty list

lethiciars committed:
Update build.gradle mainClassName

lethiciars committed:
Make Task class abstract

lethiciars committed:
Fix exit feature
